packages/cdktf/lib/terraform-asset.ts

```diff
@@ -0,0 +1,102 @@
+import { Construct, ISynthesisSession, Node } from "constructs";
+import * as fs from "fs";
+import * as path from "path";
+import { Manifest } from "./manifest";
+import { copySync, hashPath } from "./private/fs";
+import { Resource } from "./resource";
+
+export interface TerraformAssetConfig {
+  readonly path: string;
+  readonly type?: AssetType;
+  readonly assetHash?: string;
```

Comment:
What scenario do you see where a user would pass this configuration in?

Reply:
A user could choose a hash based on their needs, e.g. ignoring node modules to have more fine grained control around recreation of resources